clvm/operators.py

```diff
@@ -35,6 +49,97 @@
 }
 
 
+def args_len(op_name, args):
+    for arg in args.as_iter():
+        if arg.pair:
+            raise EvalError("%s requires int args" % op_name, arg)
+        yield len(arg.as_atom())
+
+
+# unknown ops are reserved if they start with 0xffff
+# otherwise, unknown ops are no-ops, but they have costs. The cost is computed
+# like this:
+
+# byte index (reverse):
+# n | .... | 1        | 0        |
+# --+- - --+------+---+----------+
+# n | .... |      |fun| ignored  |
+# --+- - --+------+---+----------+
+# multiplier      |
+#                 | 2 bits
+
+# 1 is always added to the multiplier before using it to multiply the cost, this
+# is since cost may not be 0.
+
+# fun is 2 bits and defines how cost is computed based on arguments:
+# 0: constant, cost is 1 * (multiplier + 1)
+# 1: computed like operator add, multiplied by (multiplier + 1)
+# 2: computed like operator mul, multiplied by (multiplier + 1)
+# 3: computed like operator concat, multiplied by (multiplier + 1)
+
+# this means that unknown ops where fun is 1, 2, or 3, may still be fatal errors
+# if the arguments passed are not atoms.
+
+def default_unknown_op(op: bytes, args: CLVMObject) -> Tuple[int, CLVMObject]:
+    # any opcode starting with ffff is reserved (i.e. fatal error)
+    if len(op) >= 2 and op[:2] == b"\xff\xff":
+        raise EvalError("reserved operator", args.to(op))
+
+    # all other unknown opcodes are no-ops
+    # the cost of the no-ops is determined by the opcode number, except the
+    # least significant byte. unknown opcodes of 1 byte have cost 1.
+    if len(op) <= 1:
+        return (1, SExp.null())
+
+    cost_function = op[-2] & 0x3
+    # the multiplier cannot be 0. it starts at 1
+    cost_multiplier = (int.from_bytes(op[:-1], "big", signed=False) >> 2) + 1
+
+    # 0 = constant
+    # 1 = like op_add/op_sub
+    # 2 = like op_multiply
+    # 3 = like op_concat
+    if cost_function == 0:
+        cost = 1
```

Comment:
> cost = 1

I believe we have a MIN_COST global floating around somewhere. May want to use that instead of `1`

Reply:
I wanted to keep this minimum cost to the actual absolute minimum, to leave the most doors open.